include/oboe/FullDuplexStream.h

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef OBOE_FULL_DUPLEX_STREAM_
+#define OBOE_FULL_DUPLEX_STREAM_
+
+#include <cstdint>
+#include "oboe/Definitions.h"
+#include "oboe/AudioStream.h"
+#include "oboe/AudioStreamCallback.h"
+
+namespace oboe {
+
+class FullDuplexStream : public AudioStreamCallback {
+public:
+    FullDuplexStream() {}
+    virtual ~FullDuplexStream() = default;
+
+    void setInputStream(AudioStream *stream) {
+        mInputStream = stream;
+    }
+
+    AudioStream *getInputStream() {
+        return mInputStream;
+    }
+
+    void setOutputStream(AudioStream *stream) {
+        mOutputStream = stream;
+    }
+    AudioStream *getOutputStream() {
+        return mOutputStream;
+    }
+
+    virtual Result start() {
```

Comment:
Add some auto docs.
This starts both input and output streams.

Reply:
Thanks!